### ISSUE TITLE
DataSet.detach() JavaDoc is wrong, it is copy-pasted from .migrate()

### ISSUE THREAD

**Issue by stolsvik** [eclipse/deeplearning4j#6860]
```
    /**
     * This method migrates this DataSet into current Workspace (if any)
     */
    @Override
    public void detach() {
        if (features != null)
            features = features.detach();

        if (labels != null)
            labels = labels.detach();

        if (featuresMask != null)
            featuresMask = featuresMask.detach();

        if (labelsMask != null)
            labelsMask = labelsMask.detach();
    }
```
